sympy/printing/codeprinter.py

```diff
@@ -14,6 +16,32 @@
 from sympy.codegen.ast import Assignment
 
 
+def prints_statement(method):
+    """ Decorator for methods printing statements (as opposed to expressions) """
+
+    @wraps(method)
+    def wrapper(self, expr):
+        if self._settings.get('enable_statements', True):
+            return method(self, expr)
+        else:
+            raise ValueError("The code-printer setting 'enable_statements' is False,\n"
+                             "but %s generates a statement in the chosen language." %
+                             type(expr))
```

Comment:
Maybe this is too pedantic, but expressions themselves are valid statements in many languages (such as Python).